test/GoodGhosting.test.js

```diff
@@ -530,6 +535,19 @@ contract("GoodGhosting", (accounts) => {
                 "FundsRedeemedFromExternalPool event should be emitted when funds are redeemed from external pool",
             );
         });
+
+        it("transfers principal to the user in case no one wins", async () => {
+            await joinGamePaySegmentsAndIncomplete(player1);
+            await goodGhosting.redeemFromExternalPool({ from: player1 });
+            const result = await goodGhosting.withdraw({ from: player1 });
+
+            truffleAssert.eventEmitted(
+                result,
+                "Withdrawal",
+                (ev) => ev.player === player1,
+                "Withdrawal event should be emitted when user tries to withdraw their principal",
+            );
+        })
```

Comment:
we should probably validate that the amount in the event equals the user's principal

Reply:
I checked this in the console it was correct but yeah we can add